packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/modifiers/iam-role-permission-boundary-modifier.ts

```diff
@@ -0,0 +1,15 @@
+import { getPermissionBoundaryArn } from 'amplify-cli-core';
+import Role from 'cloudform-types/types/iam/role';
+import { ResourceModifier } from '../pre-push-cfn-modifier';
+
+export const iamRolePermissionBoundaryModifier: ResourceModifier<Role> = async resource => {
+  if (resource?.Properties?.PermissionsBoundary) {
```

Comment:
the transformer isn't meant to be a CFN validator. In valid CFN it can only be a string so at this point I think we can assume if it exists at all we should leave it alone